doc/source/train/distributed-pytorch/experiment-tracking.rst

```diff
@@ -13,45 +13,135 @@ Ray Train also exposes logging callbacks that automate some of these tasks.
-
-2. When using the native libraries, you should report the results to both Ray Train and
-the experiment tracking library.
-
-Example:
-
-.. code-block:: python
-
-    from ray import train
-
-    def train_fn(config):
-        context = train.get_context()
-
-        wandb.init(
-            id=context.get_trial_id(),
-            name=context.get_trial_name(),
-            group=context.get_experiment_name(),
-            # ...
-        )
-        # ...
-
-        loss = optimize()
-
-        metrics = {"loss": loss}
-        # Only report the first worker results to wandb
-        if context.get_world_rank() == 0:
-            wandb.log(metrics)
-
-        # Also report to Ray Train. Note that this _must_ happen for all workers.
-        train.report(metrics)
+    .. tab-item:: PyTorch
 
+        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
+        Comet directly in your Ray Train training loop.
+
+        There are two things to keep in mind:
 
+        1. Your code is executed in parallel on many workers. However, you often only want to report
+        results from one of these workers (usually the first worker - the "rank 0" worker).
+
+        2. When using the native libraries, you should report the results to both Ray Train and
+        the experiment tracking library.
+
+        Example:
+
+        .. code-block:: python
+
+            from ray import train
+
+            def train_fn(config):
+                context = train.get_context()
+
+                wandb.init(
+                    id=context.get_trial_id(),
+                    name=context.get_trial_name(),
+                    group=context.get_experiment_name(),
+                    # ...
+                )
+                # ...
+
+                loss = optimize()
+
+                metrics = {"loss": loss}
+                # Only report the first worker results to wandb
+                if context.get_world_rank() == 0:
+                    wandb.log(metrics)
+
+                # Also report to Ray Train. Note that this _must_ happen for all workers.
+                train.report(metrics)
+
+    .. tab-item:: PyTorch Lightning
+
+        You can keep using the Lightning's native Logger integrations in the worker 
+        function. They should work out-of-the box with Ray TorchTrainer.
+
+        Example:
+
+        .. code-block:: python
+            
+            import pytorch_lightning as pl
+            from pytorch_lightning.loggers.wandb import WandbLogger
+            from pytorch_lightning.loggers.comet import CometLogger
+            from pytorch_lightning.loggers.mlflow import MLFlowLogger
+
+            def train_func_per_worker():
+                ...
+
+                wandb_logger = WandbLogger(
+                    name="demo-run", 
+                    project="demo-project", 
+                    id="unique_id",  
+                    offline=offline
+                )
```

Comment:
Quick question, does this work out of the box with multiple parallel workers? Does Lightning make sure only rank 0 reports results?

Reply:
yes, Lightning itself ensures wandb session only created on rank 0 worker.